include/camp/tuple.hpp

```diff
@@ -359,14 +359,21 @@ class tagged_tuple : public tuple<Elements...>
 };
 
 template <>
-class tuple<>
+struct tuple<>
 {
 public:
   using TList = camp::list<>;
   using TMap = TList;
   using type = tuple;
 };
 
+#if defined(__cplusplus) && __cplusplus >= 201703L
+/// Class template argument deduction rule for tuples
+/// e.g. camp::tuple t{1, 2.0};
+template <typename... T>
+tuple(T...) -> tuple<T...>;
```

Comment:
Mind adding one for tagged tuple while you're in here? Glad to have this done, thanks for putting it up!

Reply:
Sure! Done!

Reply:
Actually, I still need to add tests.

Reply:
Cool, send me a ping when you're happy with the state and I'll take another look, so far this is all good with me.

Reply:
I'm not actually sure if we can provide CTAD for tagged_tuple since there's no constructor that takes the TagList. And structured binding is a little weird since the TagList also has to be pulled out.

Reply:
Oh, but if the TagList is also pulled out, there's an out of range assert.

Reply:
OK, I got structured bindings to work with tagged_tuple. I'm still attempting to figure out the CTAD.

Reply:
@trws, it looks like partial CTAD is what we would need and that is not supported in the standard. Either that, or we need a constructor that takes the TagList as well. Thoughts?